test/e2e/set/scyllacluster/scyllamanager_object_storage.go

```diff
@@ -779,13 +805,49 @@ var _ = g.Describe("Scylla Manager integration", framework.RequiresObjectStorage
 		o.Expect(backupTask.Properties).To(o.Equal(previousBackupTask.Properties))
 
 		previousRepairTask := repairTask
-		repairTasks, err = managerClient.ListTasks(ctx, *sc.Status.ManagerID, "repair", false, "", "")
+		repairTasks, err = managerClient.ListTasks(ctx, managerClusterID, "repair", false, "", "")
 		o.Expect(err).NotTo(o.HaveOccurred())
 		o.Expect(repairTasks.TaskListItemSlice).To(o.HaveLen(1))
 		repairTask = repairTasks.TaskListItemSlice[0]
 		o.Expect(repairTask.Name).To(o.Equal(sc.Status.Repairs[0].Name))
 		o.Expect(repairTask.ID).To(o.Equal(*sc.Status.Repairs[0].ID))
 		o.Expect(repairTask.Properties).To(o.Equal(previousRepairTask.Properties))
+
+		framework.By("Deleting ScyllaCluster")
+		err = f.ScyllaClient().ScyllaV1().ScyllaClusters(sc.Namespace).Delete(
+			ctx,
+			sc.Name,
+			metav1.DeleteOptions{
+				PropagationPolicy: pointer.Ptr(metav1.DeletePropagationForeground),
+				Preconditions: &metav1.Preconditions{
+					UID: &sc.UID,
+				},
+			})
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		framework.By("Waiting for ScyllaCluster to be deleted")
+		deletionCtx, deletionCtxCancel := context.WithTimeout(ctx, utils.SyncTimeout)
+		defer deletionCtxCancel()
+		err = framework.WaitForObjectDeletion(
+			deletionCtx,
+			f.DynamicClient(),
+			scyllav1.GroupVersion.WithResource("scyllaclusters"),
+			sc.Namespace,
+			sc.Name,
+			&sc.UID,
+		)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		framework.By("Verifying that the cluster was removed from the global ScyllaDB Manager state")
+		o.Eventually(func(eo o.Gomega, ctx context.Context) {
+			_, err := managerClient.GetCluster(ctx, managerClusterID)
+			eo.Expect(err).To(o.HaveOccurred())
+			eo.Expect(err).To(o.Satisfy(managerclienterrors.IsNotFound))
+		}).
+			WithContext(ctx).
+			WithTimeout(utils.SyncTimeout).
+			WithPolling(5 * time.Second).
+			Should(o.Succeed())
```

Comment:
Couldn't we extract a helper deleting a `ScyllaCluster` and verifying its deletion? We've got this duplicated in `Scylla Manager integration` test.

Reply:
Added a separate test for registration/deregistration.

Reply:
Makes sense. :+1: I like the proposed approach as it promotes a "single reason to fail" rule of thumb for tests.

Reply:
https://github.com/scylladb/scylla-operator/issues/2707